src/test/java/io/confluent/copycat/hdfs/HdfsWriterTest.java

```diff
@@ -123,19 +124,46 @@ public void testWriteRecordNonZeroInitailOffset() throws IOException, CopycatExc
 
     long[] validOffsets = {2, 5, 8, 9};
     for (int i = 1; i < validOffsets.length; i++) {
-      Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, validOffsets[i]));
+      long startOffset = validOffsets[i - 1] + 1;
+      long endOffset = validOffsets[i];
+      Path
+          path =
+          new Path(FileUtils.committedFileName(url, topicsDir, topicPart, startOffset, endOffset));
       Collection<Object> records = readAvroFile(path);
-      long size = validOffsets[i] - validOffsets[i - 1];
+      long size = endOffset - startOffset + 1;
       assertEquals(records.size(), size);
-      for (Object avroRecord: records) {
+      for (Object avroRecord : records) {
         assertEquals(avroRecord, AvroData.convertToAvro(record));
       }
     }
+  }
+
+  @Test
+  public void testTimedRotate() throws IOException, CopycatException, InterruptedException {
+    Properties props = createProps();
+    HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
+    HdfsWriter hdfsWriter = new HdfsWriter(connectorConfig);
+
+    String topicsDir = connectorConfig.getString(HdfsSinkConnectorConfig.TOPIC_DIR_CONFIG);
+    String topic = "topic";
+    int partition = 0;
+    String key = "key";
+    GenericRecord record = createRecord();
+    TopicPartition topicPart = new TopicPartition(topic, partition);
+
+    for (long offset = 0; offset < 2; offset++) {
+      SinkRecord sinkRecord = new SinkRecord(topic, partition, key, record, offset);
+      hdfsWriter.writeRecord(topicPart, sinkRecord);
+    }
+
+    Thread.sleep(11000);
+    hdfsWriter.close();
+
+    Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, 0, 1));
+    Collection<Object> records = readAvroFile(path);
 
-    long[] inValidOffsets = {3, 4, 6, 7};
-    for (long offset: inValidOffsets) {
-      Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, offset));
-      assertFalse(fs.exists(path));
+    for (Object avroRecord : records) {
+      assertEquals(avroRecord, AvroData.convertToAvro(record));
```

Comment:
1. Will add those tests
2. We can probably create a WAL with some entries and create some temp files. We can test when the recovery process can properly commit the files in WAL.
3. Do you mean that we have the DFSClient to lose connectivity to HDFS so that a client can lose lease on a file?
4. I tested this. The WAL should work properly in this case. I can add this test to WALtest.